### PR TITLE
[Wildcard] Migrate remaining `data-tooltip`s

### PR DESCRIPTION
**Pull request by lrhacker**
This PR removes all other outstanding `data-tooltip` uses currently present on `main` (excluding the one in InputTooltip, which is handled by #38822).

Also, fixes tests that were still using `data-tooltip` property in conjunction with `setElementTooltip` (for code hosts) - the actual implementation of `setElementTooltip` in this case uses `aria-label`, not `data-tooltip`, so the tests now reflect that.

## Test plan

Will need to verify all these replaces tooltips still look and behave as expected.

## App preview:

- [Web](https://sg-web-lrh-tooltip-migrations-final.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-roaxpjnzhi.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/repo/RepoRevisionSidebarSymbols.tsx [hunk 1]
```diff
@@ -48,8 +48,10 @@ const SymbolNode: React.FunctionComponent<React.PropsWithChildren<SymbolNodeProp
                 activeClassName={styles.linkActive}
                 onClick={onHandleClick}
             >
-                <SymbolIcon kind={node.kind} className="mr-1 test-symbol-icon" />
-                <span className={classNames('test-symbol-name', styles.name)}>{node.name}</span>
+                <SymbolIcon kind={node.kind} className="mr-1" />
```
Comment (jasongornall) [inline client/web/src/repo/RepoRevisionSidebarSymbols.tsx:51]: guessing test-symbol-icon didn't do anything?

File: client/vscode/src/webview/search-panel/alias/RepoFileLink.tsx [hunk 2]
```diff
@@ -87,20 +87,19 @@ export const RepoFileLink: React.FunctionComponent<React.PropsWithChildren<Props
...
-            </button>{' '}
-            ›{' '}
-            <button onClick={onFileClick} type="button" className="btn btn-text-link">
-                {fileBase ? `${fileBase}/` : null}
-                <strong>{fileName}</strong>
-            </button>
-        </div>
+        <Tooltip content={truncated ? (fileBase ? `${fileBase}/${fileName}` : fileName) : null}>
+            <span>
+                <div ref={titleReference} className={className} onMouseEnter={checkTruncation}>
```
Comment (jasongornall) [inline client/vscode/src/webview/search-panel/alias/RepoFileLink.tsx:92]: tricky tricky

**Review by jasongornall — APPROVED** [review]

**Review by jasongornall — APPROVED** [review]

File: client/web/src/repo/RepoRevisionSidebarSymbols.tsx (hunk 1 above)
Comment (lrhacker) [reply client/web/src/repo/RepoRevisionSidebarSymbols.tsx:51]: Changed it to `data-testid="symbol-icon"` in the `SymbolIcon` component 🙂

**Review by umpox — APPROVED** [review]
🚀

**Pull request merged**